client/src/components/Chat/Messages/Content/ToolCallInfo.tsx

```diff
@@ -57,16 +73,27 @@ export default function ToolCallInfo({
   // Extract ui_resources from the output to display them in the UI
   let uiResources: UIResource[] = [];
   if (output?.includes('ui_resources')) {
-    const parsedOutput = JSON.parse(output);
-    const uiResourcesItem = parsedOutput.find(
-      (contentItem) => contentItem.metadata === 'ui_resources',
-    );
-    if (uiResourcesItem?.text) {
-      uiResources = JSON.parse(atob(uiResourcesItem.text)) as UIResource[];
+    try {
```

Comment:
@sbruel you said [here](https://github.com/danny-avila/LibreChat/pull/9299#discussion_r2310684499):

> we probably need more solid error handling if the output includes ui_resources as text, for some reason, without being a proper ui_resource

This case is actually properly handled, and there's now a test case proving it: 

https://github.com/danny-avila/LibreChat/blob/964c973fa1c74d7ebe66c48150e77bcd0bb76244/client/src/components/Chat/Messages/Content/__tests__/ToolCallInfo.test.tsx#L171-L191.